src/simplify_algebra.cpp

```diff
@@ -266,6 +266,84 @@ struct find_mul_dot
     }
 };
 
+struct find_dot_slice
+{
+    auto matcher() const
+    {
+        return match::name("slice")(
+            match::args(match::name("dot", "quant_dot")(match::used_once()).bind("dot_ins")));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto slice_ins = r.result;
+        auto dot_ins   = r.instructions["dot_ins"];
+        auto slice_op  = slice_ins->get_operator().to_value();
+        auto axes      = slice_op["axes"].to_vector<int64_t>();
+        auto starts    = slice_op["starts"].to_vector<int64_t>();
+        auto ends      = slice_op["ends"].to_vector<int64_t>();
+        assert(starts.size() == ends.size() and starts.size() == axes.size());
+        auto has_neg_vals = [](auto vec) {
+            return std::any_of(vec.begin(), vec.end(), [](auto i) { return i < 0; });
+        };
+        if(has_neg_vals(starts) or has_neg_vals(ends) or has_neg_vals(axes))
+        {
+            return;
+        }
```

Comment:
Maybe this should be a throw or a warning because having a negative value here would probably be a problem.

Reply:
We have `normlize_ops` pass which runs at the beginning which would convert all the neg axes to positive ones. 
Therefore i think retuning simply is okay.

Reply:
I think it should print a warning if possible, otherwise if we mess something up wrt. `normalize_ops` this could be passed over without us knowing.

Reply:
made changes to use "normalized_operator" and also enabled test for neg vals.